react-components/select/step4.md

```diff
@@ -0,0 +1,39 @@
+Customize the `select` component. There are many options available to customize the `select` component and its options.
+
+One such customization, for both typeahead and multiple typeahead variants, is custom filtering.
+
+By default, these variants match the entered input with the options' values in order. For example, given the "Alabama" option value, an "Ala" text input would match but a "bama" text input would not.
+
+Allow any input to be matched, regardless of ordering.
+
+1) **Add the following custom filtering function to the constructor of the App class.**
+
+The component expects a function that will return a list of options to display. In the code below, the text input is matched regardless of ordering using a regular expression, and sanitizes the text input to avoid extraneous characters. Using the same example above, this will mean that a text input of "bama" would still match for the option "Alabama".
```

Comment:
```suggestion
The component expects a function that will return a list of options to display. In the code below, the text input is matched regardless of ordering using a regular expression and sanitizes the text input to avoid extraneous characters. Using the same example above, this will mean that a text input of "bama" would still match for the option "Alabama".
```